lib/opensusebasetest.pm

```diff
@@ -359,38 +365,26 @@ sub export_logs_desktop {
 
     if (check_var("DESKTOP", "kde")) {
         if (get_var('PLASMA5')) {
-            my $fn  = '/tmp/plasma5_configs.tar.bz2';
-            my $cmd = sprintf 'tar cjf %s /home/%s/.config/*rc', $fn, $username;
-            type_string "$cmd\n";
-            upload_logs $fn;
+            $self->tar_and_upload_log("/home/$username/.config/*rc", '/tmp/plasma5_configs.tar.bz2');
         }
         else {
-            my $fn  = '/tmp/kde4_configs.tar.bz2';
-            my $cmd = sprintf 'tar cjf %s /home/%s/.kde4/share/config/*rc', $fn, $username;
-            type_string "$cmd\n";
-            upload_logs $fn;
+            $self->tar_and_upload_log("/home/$username/.kde4/share/config/*rc", '/tmp/kde4_configs.tar.bz2');
         }
         save_screenshot;
     } elsif (check_var("DESKTOP", "gnome")) {
-        script_run("tar -jcv -f /tmp/gdm.tar.bz2  /home/bernhard/.cache/gdm");
-        upload_logs('/tmp/gdm.tar.bz2', failok => 1);
+        $self->tar_and_upload_log('/home/bernhard/.cache/gdm', '/tmp/gdm.tar.bz2');
     }
 
     # check whether xorg logs exist in user's home, if yes, upload xorg logs
     # from user's home instead of /var/log
-    my $log_path = '/home/*/.local/share/xorg/X*';
-    if (!script_run("ls -l $log_path")) {
-        $self->save_and_upload_log("cat $log_path", '/tmp/Xlogs.users.log', {screenshot => 1});
+    my $log_path = '/home/*/.local/share/xorg/';
+    if (!script_run("test -d $log_path")) {
```

Comment:
I guess that by this point we're already root by this point

Reply:
We are root in the post_fail_hook, not in the failed module, eg. gedit, chrome.

Reply:
Well´'didn't fail before, so... no reason to fail now :) ![](https://media.giphy.com/media/dSNaWyZiwyRdm/giphy.gif)